lib/vmpooler/prometheus.rb

```diff
@@ -0,0 +1,240 @@
+# frozen_string_literal: true
+
+module Vmpooler
+  class VmpPrometheus
+    attr_reader :prefix
+
+    # Constants for Metric Types
+    M_COUNTER   = 1
+    M_GAUGE     = 2
+    M_SUMMARY   = 3
+    M_HISTOGRAM = 4
+
+    @p_metrics = {}
+
+    # Metrics structure used to register the metrics and also translate/interpret the incoming metrics.
+    def vmpooler_metrics_table
+      {
+        errors: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler errors',
+          prom_metric_prefix: 'vmpooler_errors',
+          metric_suffixes: %i[markedasfailed duplicatehostname],
+          param_labels: %i[template_name]
+        },
+        usage: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler usage',
+          prom_metric_prefix: 'vmpooler_usage',
+          param_labels: %i[user poolname]
+        },
+        user: {
+          mtype: M_COUNTER, # This is going to challenge prometheus
```

Comment:
If you are going to leave this note in here please expand on why it will challenge prometheus

Reply:
Done